tests/integration/tls_tests/test_tls.py

```diff
@@ -67,18 +65,14 @@ async def test_rotate_tls_key(ops_test: OpsTest) -> None:
             ops_test, unit.name, DB_SERVICE
         )
 
-    with open(PRIVATE_KEY_PATH) as f:
-        key_contents = f.readlines()
-        key_contents = "".join(key_contents)
-
     # set external and internal key using auto-generated key for each unit
     for unit in ops_test.model.applications[DATABASE_APP_NAME].units:
         action = await unit.run_action(action_name="set-tls-private-key")
         action = await action.wait()
         assert action.status == "completed", "setting external and internal key failed."
 
-    # sleep for long enough to process new certificate request and update certs
-    sleep(30)
+    # wait for certificate to be available and processed.
+    await ops_test.model.wait_for_idle()
```

Comment:
Should this be pointing at a specific application?

Reply:
this waits for all applications in the model to be idle , so its the same as `applications=[APP_1, APP_1]`